indigo/indigo-extras/src/main/scala/indigoextras/waypoints/WaypointPath.scala

```diff
@@ -0,0 +1,113 @@
+package indigoextras.waypoints
+
+import indigo.Radians
+import indigo.Vector2
+
+import scala.annotation.tailrec
+
+trait WaypointPath:
+  def waypoints: List[Vector2]
```

Comment:
I shall forever wonder if this is the right thing to have done, but for consistency, `Vector2` should probably be `Vertex`. They're almost identical, but a vertex is a position on a graph / space where vector2 is an expression of direction. `LineSegment` uses `Vertex`, I am open to counter arguments. 😄 

https://github.com/PurpleKingdomGames/indigo/blob/684df20105592b7e9e2d34529621824d584c366b/indigo/indigo/src/main/scala/indigo/shared/geometry/LineSegment.scala#L5

Reply:
Additionally, can we replace `List` with `Batch`? `Batch` is _much_ faster and used everywhere.

Reply:
I'm glad you mention this because I just realised yesterday that there is both `Vertex` and `Vector2` and I wasn't sure what was the difference 😅 I'll change both.